FIRST-26 | Auth 커스텀 어노테이션 추가

Zeeeen0 opened the pull request:
Interceptor단에서 Auth 컨트롤러 요청에 대해 JWT 토큰 검증하도록 수정

- testController.java 파일은 token값 userSeq, deviceId 사용하는 코드 임시로 넣어놓아 추후 삭제 예정입니다.

Zeeeen0 committed:
Auth 커스텀 어노테이션 추가
Interceptor단에서 Auth 컨트롤러 요청에 대해 JWT 토큰 검증하도록 수정

Zeeeen0 committed:
AccessToken 재발급 End Point 추가
userSeq Long 자료형으로 변경
Interceptor, JwtUtil 일부 리팩토링

Zeeeen0 committed:
Filter 추가 (request, response 로깅)
(Interceptor 단에서는 권한, Controller 관련 처리만 작업하도록)

sbywww commented on src/main/java/com/heygongc/global/interceptor/Interceptor.java:
```diff
@@ -31,28 +35,39 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
+        logger.info("userSeq({}), deviceId({})",userSeq, deviceId);
 
         // DB의 deviceId와 토큰값이 다를 경우
         User user = userRepository.findById(Long.valueOf(userSeq)).orElseThrow(UserNotFoundException::new);
```
bearer 인증방식으로 가는게 좋을 것 같습니다.
header 키는 authorization 으로 가는 방법으로 찾아보시면 될 것 같아요.

Zeeeen0 replied:
> bearer 인증방식으로 가는게 좋을 것 같습니다. header 키는 authorization 으로 가는 방법으로 찾아보시면 될 것 같아요.

수정했습니다. Interceptor에서는 "Authorization : Bearer {ACCESS_TOKEN}" 으로 받아서, 토큰 검증하도록 했어요.

sbywww commented on src/main/java/com/heygongc/testController.java:
```diff
@@ -13,6 +14,7 @@
+    @Auth
     @PostMapping("/getUserAndDevice")
     public ResponseEntity<String> sendUser(HttpServletRequest request) {
         String userSeq = (String) request.getAttribute("userSeq");
```
springboot 3.1부터 지원하는 HandlerMethodArgumentResolver 사용해서 구현하시는게 좋을 것 같습니다.

Zeeeen0 replied:
> springboot 3.1부터 지원하는 HandlerMethodArgumentResolver 사용해서 구현하시는게 좋을 것 같습니다.

LoginUserArgumentResolver 생성하여 userSeq, deviceId 세팅한 User 객체 넘겨주도록 추가했습니다.

Zeeeen0 committed:
HandlerMethodArgumentResolver 추가
Interceptor - request에 userSeq, deviceID값 세팅 삭제 / refreshToken 검증 삭제

Zeeeen0 committed:
Service - TokenResponse 대신 AuthToken return하도록 수정
Controller - AuthToken 받아서 new TokenResponse return하도록 수정
User Custom Exception 추가

Zeeeen0 committed:
HandlerMethodArgumentResolver에서 토큰값이 아닌 실제 DB에서 User 객체 return해주도록 수정

Zeeeen0 committed:
회원탈퇴 Boolean -> void 수정

Zeeeen0 committed:
UserException을 Interceptor와 동일하게 401, 403 Exception으로 수정

Zeeeen0 committed:
1. Swagger API 설정
2. 로그인 프로세스 수정
 - AS-IS 성공(200 OK with accessToken) / 회원가입 필요(200 OK null)
 - TO-BE 성공(200 OK with accessToken) / 회원가입 필요(204 No Content)

sbywww reviewed (approved):
goods!
